pyrobosim/test/utils/test_pose_utils.py

```diff
@@ -259,16 +348,111 @@ def test_get_matrices():
     assert pose.get_translation_matrix() == pytest.approx(expected_translation_matrix)
 
 
-def test_is_approx():
+@pytest.mark.parametrize(
+    "pose1, pose2, rel_tol, abs_tol, expected_result",
+    [
+        (
+            Pose(
+                x=1.0,
+                y=2.0,
+                z=3.0,
+                roll=np.pi / 2,
+                pitch=0.0,
+                yaw=-np.pi / 2,
+                angle_units="radians",
+            ),
+            Pose(
+                x=1.0 + 1e-4,
+                y=2.0,
+                z=3.0,
+                roll=np.pi / 2,
+                pitch=1e-4,
+                yaw=-np.pi / 2,
+                angle_units="radians",
+            ),
+            1e-3,
+            1e-3,
+            True,
+        ),
+        (
+            Pose(
+                x=1.0,
+                y=2.0,
+                z=3.0,
+                roll=np.pi / 2,
+                pitch=0.0,
+                yaw=-np.pi / 2,
+                angle_units="radians",
+            ),
+            Pose(
+                x=1.0 + 1e-4,
+                y=2.0,
+                z=3.0,
+                roll=np.pi / 2,
+                pitch=1e-4,
+                yaw=-np.pi / 2,
+                angle_units="radians",
+            ),
+            1e-5,
+            1e-5,
+            False,
+        ),
+        (
+            Pose(
+                x=1.0,
+                y=2.0,
+                z=3.0,
+                roll=90.0,
+                pitch=0.0,
+                yaw=-90.0,
+                angle_units="degrees",
+            ),
+            Pose(
+                x=1.0 + 1e-4,
+                y=2.0,
+                z=3.0,
+                roll=90.0,
+                pitch=1e-4,
+                yaw=-90.0,
+                angle_units="degrees",
+            ),
+            1e-3,
+            1e-3,
+            True,
+        ),
+        (
+            Pose(
+                x=1.0,
+                y=2.0,
+                z=3.0,
+                roll=90.0,
+                pitch=0.0,
+                yaw=-90.0,
+                angle_units="degrees",
+            ),
+            Pose(
+                x=1.0 + 1e-4,
+                y=2.0,
+                z=3.0,
+                roll=90.0,
+                pitch=1e-4,
+                yaw=-90.0,
+                angle_units="degrees",
+            ),
+            1e-5,
+            1e-5,
+            False,
+        ),
+    ],
+)
+def test_is_approx(pose1, pose2, rel_tol, abs_tol, expected_result):
```

Comment:
This is using way too many lines to be readable. 

I think in this particular test, it may be better just to have 2 different tests for each type of angle unit.